crates/core/src/db/update.rs

```diff
@@ -182,14 +182,21 @@ fn equiv(a: &TableDef, b: &TableDef) -> bool {
         table_type,
         table_access,
     } = a;
+
+    // Constraints may not be in the same order, depending on whether the
+    // `TableDef` was loaded from the db catalog or the module.
+    fn as_set(constraints: &[ConstraintDef]) -> BTreeSet<&ConstraintDef> {
+        constraints
+            .iter()
+            .filter(|c| c.constraints != Constraints::unset())
+            .collect()
+    }
+
```

Comment:
Could be better to do this change inside `TableDef`?:
```rust
pub struct TableDef {
    pub table_name: String,

    pub constraints: BtreeSet<ConstraintDef>,
}
```

Reply:
Do we actually want to store them like that? We might just want this function as a way to compare them. I think I'm going to propose that we merge this and consider that for future work.

Reply:
It threw me off a bit to realize that there can be unset constraints in the table def, so we would still need an element-wise comparison if it was a set already.